src/components/icon/icon.spec.js

```diff
@@ -109,6 +112,55 @@ describe('Icon component', () => {
     });
   });
 
+  describe('custom colors', () => {
+    const correctColors = ['red', 'slateShade50', 'rgb(0,123,100)', 'hsl(0,100%,50%)', '#123456'];
+    describe.each(correctColors)('when color prop is provided', (color) => {
+      it('takes precedence over iconColor and renders properly colored Icon', () => {
+        const wrapper = mount(<Icon color={ color } />);
+        assertStyleMatch({
+          color: toColor(baseTheme, color)
+        }, wrapper.find(StyledIcon));
+      });
+
+      it('renders properly colored Icon when hovered', () => {
+        const wrapper = mount(<Icon color={ color } />);
+        assertStyleMatch({
+          color: shade(0.2, (toColor(baseTheme, color)))
+        }, wrapper.find(StyledIcon), { modifier: ':hover' });
+      });
+    });
+    describe.each(correctColors)('when bg prop is provided', (color) => {
+      it('takes precedence over bgTheme and renders properly colored Icon', () => {
+        const wrapper = mount(<Icon bg={ color } />);
+        assertStyleMatch({
+          backgroundColor: toColor(baseTheme, color)
+        }, wrapper.find(StyledIcon));
+      });
+
+      it('renders properly colored Icon when hovered', () => {
+        const wrapper = mount(<Icon bg={ color } />);
+        assertStyleMatch({
+          backgroundColor: shade(0.2, (toColor(baseTheme, color)))
+        }, wrapper.find(StyledIcon), { modifier: ':hover' });
+      });
+    });
+
+    const wrongColors = ['rgb(0,0)', '#ff', 'test'];
+    describe.each(wrongColors)('when wrong color prop is provided', (color) => {
+      it('throws an error', () => {
+        jest.spyOn(global.console, 'error');
+        mount(<Icon color={ color } />);
+        expect(console.error).toHaveBeenCalled();
```

Comment:
Could you put `// eslint-disable-next-line no-console` above these `console.error` please? To stop the lint warnings